src/apps/explorer/components/TransanctionBatchGraph/index.tsx

```diff
@@ -303,16 +304,22 @@ function TransanctionBatchGraph({
   )
 
   useEffect(() => {
-    const cy = cytoscapeRef.current
-    setElements([])
-    if (error || isLoading || !networkId || !heightSize || !cy) return
-
-    setElements(getNodes(txSettlement, networkId, heightSize, layout.name))
-    if (resetZoom) {
-      updateLayout(cy, layout.name)
+    try {
+      setFailedToLoadGraph(false)
+      const cy = cytoscapeRef.current
+      setElements([])
+      if (error || isLoading || !networkId || !heightSize || !cy) return
+
+      setElements(getNodes(txSettlement, networkId, heightSize, layout.name))
+      if (resetZoom) {
+        updateLayout(cy, layout.name)
+      }
+      removePopper(cyPopperRef)
+      setResetZoom(null)
+    } catch (e) {
+      console.error(`Failed to build graph`, e)
+      setFailedToLoadGraph(true)
```

Comment:
thanks for addressing also this.

Should we have, maybe in another PR, an app-level https://legacy.reactjs.org/docs/error-boundaries.html
I thought we did. 

If we don't handle the errors, there should be a last resource boundary so the app don't look so broken (white screen)

Reply:
Yeah, that was my initial approach, using Sentry's Error boundary because there's no standard way to do that in react other than with Class based components

See [this](https://stackoverflow.com/questions/48482619/how-can-i-make-use-of-error-boundaries-in-functional-react-components) and [this](https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/error_boundaries/)

Wanted to have a more localized boundary to not break the UI completely. In the end, handling in the component was enough.

Will create an issue for handling this at the top level.

Reply:
https://github.com/cowprotocol/explorer/issues/503